File: Kanelbullar/kanelbullar.java
import java.util.Scanner;

public class kanelbullar {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int fr = scanner.nextInt();
        int yr = scanner.nextInt();
        int br = scanner.nextInt();
        int mr = scanner.nextInt();
        int sr = scanner.nextInt();

        int f = scanner.nextInt();
        int y = scanner.nextInt();
        int b = scanner.nextInt();
        int m = scanner.nextInt();
        int s = scanner.nextInt();

        int cinnamonBuns = Math.min(f / fr, Math.min(y / yr, Math.min(b / br, Math.min(m / mr, s / sr))));
        System.out.println(cinnamonBuns);
    }
}

File: Kanelbullar/kanelbullar.c
#include <stdio.h>

int main() {
    int fr, yr, br, mr, sr;
    scanf("%d", &fr);
    scanf("%d", &yr);
    scanf("%d", &br);
    scanf("%d", &mr);
    scanf("%d", &sr);

    int f, y, b, m, s;
    scanf("%d", &f);
    scanf("%d", &y);
    scanf("%d", &b);
    scanf("%d", &m);
    scanf("%d", &s);

    int cinnamonBuns = f / fr;

    if ((y / yr) < cinnamonBuns) {
        cinnamonBuns = y / yr;
    }

    if ((b / br) < cinnamonBuns) {
        cinnamonBuns = b / br;
    }

    if ((m / mr) < cinnamonBuns) {
        cinnamonBuns = m / mr;
    }

    if ((s / sr) < cinnamonBuns) {
        cinnamonBuns = s / sr;
    }

    printf("%d\n", cinnamonBuns);
    return 0;
}
